nixos/modules/virtualisation/includes-to-excludes.py

```diff
@@ -0,0 +1,70 @@
+
+# Convert a list of strings to a regex that matches everything but those strings
+# ... and it had to be a POSIX regex; no negative lookahead :(
+# This is a workaround for erofs supporting only exclude regex, not an include list
+
+import sys
+
+# We can configure this script to match in different ways if we need to.
+# The regex got too long for the argument list, so we had to truncate the
+# hashes and use matchStringPrefix. That's less accurate, and might pick up some
+# garbage like .lock files, but only if the sandbox doesn't hide those. Even
+# then it should be harmless.
+
+# Produce the negation of ^a$
+matchExactly = ".+"
+# Produce the negation of ^a
+matchStringPrefix = "//X" # //X should be epsilon regex instead. Not supported??
+# Produce the negation of ^a/?
+matchSubpaths = "[^/].*$"
+
+# matchEnd = matchSubpaths
+matchEnd = matchStringPrefix
+# matchEnd = matchExactly
+
+# There's probably at least one bug in here, but it seems to works well enough
+# for filtering store paths.
+def go(list):
+    s = "("
+
+    # Match anything that starts with the wrong letter
+    letters = dict()
+
+    for item in list:
+        if item == "":
```

Comment:
```
if not item:
  continue
```
Lets you unindent the rest.

Reply:
Thank you for your helpful suggestions.
After applying the preceding ones this one doesn't make a difference any more, as only one clause is left anyway :tada: